app/celer-g4/CMakeLists.txt

```diff
@@ -18,48 +18,48 @@ if(CELERITAS_USE_Geant4)
     TrackingAction.cc
   )
   set(LIBRARIES Celeritas::accel)
+
+  if(CELERITAS_USE_ROOT)
+    function(celeritas_local_root_generate_dictionary name)
+      # No need for the rootmap file for an executable.
+      set(CMAKE_ROOTTEST_NOROOTMAP TRUE)
+      include_directories(
+        # root_generate_dictionary cannot handle the '$<INSTALL_INTERFACE:include>'
+        # part of the Celeritas::corecel target's include, so we cannot use it in
+        # the 'DEPENDENCIES' property.
+        "${CELERITAS_HEADER_CONFIG_DIRECTORY}"
+        "${PROJECT_SOURCE_DIR}/src"
+        # The include directory are attached to the Geant4 target through
+        # INTERFACE_INCLUDE_DIRECTORIES which is not looked at by
+        # root_generate_dictionary.
+        ${Geant4_INCLUDE_DIRS}
+        ${CLHEP_INCLUDE_DIRS}
+      )
+      root_generate_dictionary(${name}
+        ${CMAKE_CURRENT_SOURCE_DIR}/HitRootIO.hh
+        ${CMAKE_CURRENT_SOURCE_DIR}/SensitiveHit.hh
+        NOINSTALL
+        MODULE celer-g4
+        # TODO Uncomment this line and remove the include_directories when
+        # the minimal required ROOT version is v6.28/02 or later.
+        # DEPENDENCIES ${Geant4_LIBRARIES} Celeritas::accel Celeritas::corecel
+        LINKDEF "${CMAKE_CURRENT_SOURCE_DIR}/HitClassesLinkDef.h"
+      )
+    endfunction(celeritas_local_root_generate_dictionary)
+    celeritas_local_root_generate_dictionary(HitClassesRootInterfaces)
+    list(APPEND SOURCES
+      "${CMAKE_CURRENT_BINARY_DIR}/HitClassesRootInterfaces.cxx"
+      HitRootIO.cc
+    )
+    list(APPEND LIBRARIES ROOT::Tree)
+  endif()
 else()
   set(SOURCES
     celer-g4.nogeant.cc
   )
   set(LIBRARIES)
 endif()
 
-if(CELERITAS_USE_ROOT)
```

Comment:
The change set might be smaller if you just change this line to `CELERITAS_USE_ROOT AND CELERITAS_USE_Geant4`?

Reply:
My thinking was that the `CELERITAS_USE_Geant4` is the fundamental conditional, so seemed logical to put the "use ROOT" inside of this rather than a second dual conditional later. However, if you prefer to keep changesets small, I'll update!

Reply:
True. I'm also partial to reducing the level of indentation 😅

Reply:
That's also a good reason - o.k., will change to the `CELERITAS_USE_ROOT AND CELERITAS_USE_Geant4` form in a fresh commit, which can then be squashed on merge.